pallets/gear/src/internal.rs

```diff
@@ -586,15 +595,44 @@ where
             // Message is going to be inserted into mailbox.
             //
             // No hold bound checks required, because gas_limit isn't less than threshold.
-            if gas_limit >= threshold {
+            to_mailbox = gas_limit >= threshold;
+            if to_mailbox {
                 // Cutting gas for storing in mailbox.
-                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_limit)
+                let gas_amount = gas_for_delay.saturating_add(gas_limit);
+                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_amount)
                     .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
 
                 // TODO: adapt this line if gasful sending appears for reservations (#1828)
                 if let Some(reservation_id) = reservation {
                     Self::remove_gas_reservation_with_task(dispatch.source(), reservation_id);
```

Comment:
reservation now should disappear in all cases if Some

Reply:
also test coverage pls

Reply:
done, covered with test by adding extra checks to implemented in this PR tests